modules/vebal/root-gauges.subgraph.ts

```diff
@@ -0,0 +1,55 @@
+import { Chain } from '@prisma/client';
+import { keyBy } from 'lodash';
+
+import { RootGauge } from './root-gauges.onchain';
+import { mainnetNetworkConfig } from '../network/mainnet';
+import { GraphQLClient } from 'graphql-request';
+import { getSdk } from '../subgraphs/gauge-subgraph/generated/gauge-subgraph-types';
+
+type SubGraphRootGauge = {
+    gaugeAddress: string;
+    chain: Chain;
+    recipient?: string;
+};
+
+//TODO: use generic gauge-subgraph-service instead
+const subgraphClient = getSdk(new GraphQLClient(mainnetNetworkConfig.data.subgraphs.gauge!));
+
+export async function fetchRootGaugesFromSubgraph(onchainRootAddresses: string[]) {
+    const rootGauges = (await subgraphClient.RootGauges({ ids: onchainRootAddresses })).rootGauges;
+
+    const l2RootGauges: SubGraphRootGauge[] = rootGauges.map((gauge) => {
+        return {
+            gaugeAddress: gauge.id,
+            chain: gauge.chain.toUpperCase() as Chain,
+            recipient: gauge.recipient,
+        } as SubGraphRootGauge;
+    });
+
+    const liquidityGauges = (await subgraphClient.LiquidityGauges({ ids: onchainRootAddresses })).liquidityGauges;
+
+    const mainnetRootGauges: SubGraphRootGauge[] = liquidityGauges.map((gauge) => {
+        return {
+            gaugeAddress: gauge.id,
+            chain: Chain.MAINNET,
+            recipient: undefined,
+        } as SubGraphRootGauge;
+    });
+
+    return [...l2RootGauges, ...mainnetRootGauges];
+}
+
+export function updateOnchainGaugesWithSubgraphData(onchainGauges: RootGauge[], subgraphGauges: SubGraphRootGauge[]) {
+    const subgraphGaugesByAddress = keyBy(subgraphGauges, 'gaugeAddress');
+
+    return onchainGauges.map((gauge) => {
+        const rootGauge = gauge;
+        const subGraphGauge = subgraphGaugesByAddress[gauge.gaugeAddress];
+        if (subGraphGauge) {
```

Comment:
which root gauges are on the contract but not in the subgraph?

Reply:
We have the following ones currently: 

        '0x7aa5475b2ea29a9f4a1b9cf1cb72512d1b4ab75e', //Liquidity Mining Committee gauge
        '0x56124eb16441a1ef12a4ccaeabdd3421281b795a', // veLIT
        '0x5b79494824bc256cd663648ee1aad251b32693a9', // veUSH
        '0xb78543e00712c3abba10d0852f6e38fde2aaba4d', // veBAL
        '0x3f829a8303455cb36b7bcf3d1bdc18d5f6946aea', // Arbitrum killed gauge that was replaced by 0xd758454bdf4df7ad85f7538dc9742648ef8e6d0a
        '0xe867ad0a48e8f815dc0cda2cdb275e0f163a480b', // killed
        '0x9fb8312cedfb9b35364ff06311b429a2f4cdf422', // killed

That's why I added a check to throw in case de have a missing alive one in the future